spark/src/main/scala/org/apache/spark/sql/delta/DeltaParquetFileFormat.scala

```diff
@@ -205,16 +203,27 @@ case class DeltaParquetFileFormat(
     // TODO(SPARK-47731): Parquet reader in Spark has a bug where a file containing 2b+ rows
     // in a single rowgroup causes it to run out of the `Integer` range.
     // For Delta Parquet readers don't expose the row_index field as a metadata field.
-    if (!RowId.isEnabled(protocol, metadata)) {
-      super.metadataSchemaFields.filter(_ != ParquetFileFormat.ROW_INDEX_FIELD)
-    } else {
-      // It is fine to expose the row_index field as a metadata field when Row Tracking
-      // is enabled because it is needed to generate the Row ID field, and it is not a
-      // big problem if we use 2b+ rows in a single rowgroup, it will throw an exception and
-      // we can then use less rows per rowgroup. Also, 2b+ rows in a single rowgroup is
-      // not a common use case.
-      super.metadataSchemaFields ++ rowTrackingFields
-    }
+    val metadataSchemaFieldsWithoutRowIndexFilterFields =
+      if (!RowId.isEnabled(protocol, metadata)) {
+        super.metadataSchemaFields.filter(_ != ParquetFileFormat.ROW_INDEX_FIELD)
+      } else {
+        // It is fine to expose the row_index field as a metadata field when Row Tracking
+        // is enabled because it is needed to generate the Row ID field, and it is not a
+        // big problem if we use 2b+ rows in a single rowgroup, it will throw an exception and
+        // we can then use less rows per rowgroup. Also, 2b+ rows in a single rowgroup is
+        // not a common use case.
+        super.metadataSchemaFields ++ rowTrackingFields
+      }
+
+    val isDVsEnabled = DeltaConfigs.ENABLE_DELETION_VECTORS_CREATION.fromMetaData(metadata)
```

Comment:
`metadata` shows the current status of the table, i.e., if DV is enabled in the current version, isn't it? How about some old versions that have DV, but now the user has decided to disable it? I think we should look at the table protocol: if DV table feature is turned on, then we assume the table contains DV.

Reply:
@xupefei Quick question, can an existing table add new table feature? Like if a table does not have DV from the beginning, in the property, and then at some point later in time get added DVs.

Reply:
@xupefei I think you can do this using ALTER TABLE right? But once it is in, you cannot drop it at the moment.